src/treq/client.py

```diff
@@ -114,8 +114,8 @@ def put(self, url, data=None, **kwargs):
     def patch(self, url, data=None, **kwargs):
         return self.request('PATCH', url, data=data, **kwargs)
 
-    def post(self, url, data=None, **kwargs):
-        return self.request('POST', url, data=data, **kwargs)
+    def post(self, url, data=None, json=None, **kwargs):
```

Comment:
By my reading this will change the behavior of `treq.post(url)` (with no additional arguments). The `has_json` condition will always be `True` when this shortcut method is used, so the POST request will have a payload with `Content-Type: application/json` and an entity body of `null`.

While the current default behavior of sending an empty POST request is not particularly useful, sending JSON `null` by default seems surprising. Changing the behavior is technically backwards-incompatible. I'd suggest using a sentinel object to indicate whether the `json` keyword argument is set, or simply omitting it from the `def post()` signature.